sync/listen.py

```diff
@@ -39,26 +41,24 @@ def add_callback(self, exchange, func):
             for callbacks in self.callbacks.itervalues():
                 callbacks.append(func)
 
-    def drain_events(self, timeout=0.1):
-        """Drain all active events and call callbacks."""
-        if not self._entered:
-            raise Exception('must enter context manager before calling')
-
-        try:
-            self._conn.drain_events(timeout=timeout)
-        except socket.timeout:
-            pass
-
-    def listen_forever(self):
+    def listen_forever(self, retries=5):
         """Listen for and handle messages until interrupted."""
         if not self._entered:
             raise Exception('must enter context manager before calling')
-
-        while True:
+        pause = 1
+        while True and retries > 0:
```

Comment:
Should we reset retries on a successful connection?

Reply:
That sounds like a good idea.